src/common/row_operations/row_aggregate.cpp

```diff
@@ -81,9 +81,9 @@ void RowOperations::CombineStates(RowOperationsState &state, TupleDataLayout &la
 	idx_t offset = layout.GetAggrOffset();
 
 	for (auto &aggr : layout.GetAggregates()) {
-		D_ASSERT(aggr.function.combine);
+		D_ASSERT(aggr.function.absorb);
 		AggregateInputData aggr_input_data(aggr.GetFunctionData(), state.allocator);
-		aggr.function.combine(sources, targets, aggr_input_data, count);
+		aggr.function.absorb(sources, targets, aggr_input_data, count);
```

Comment:
Perhaps it is cleaner if we, instead of adding a new callback, add a parameter to `AggregateInputData`? For example, we can have

`AggregateCombineType::ALLOW_DESTRUCTIVE` vs `AggregateCombineType::PRESERVE_INPUT` (defaulting to `AggregateCombineType:: PRESERVE_INPUT `).